tests/scripts/test-e2e-standalone.sh

```diff
@@ -83,7 +94,18 @@ if [ $? -ne 0 ]; then
   cat "$OLS_CONFIG_FILE"
   exit 1
 fi
-set -e
 
 echo Done waiting for OLS server start, running e2e
 OLS_URL="http://localhost:8080" SUITE_ID=standalone TEST_TAGS="not cluster and not rag" make test-e2e
+rc=$?
+
+if [ ! -z ${TEST_HTTP_HEADER_REDACTION:-} ]; then
+  for h in authorization proxy-authorization cookie; do
+    curl -X 'POST' 'http://localhost:8080/v1/query' -H 'accept: application/json' \
+    -H 'Content-Type: application/json' -H "$h: random_value" \
+    -d '{"query": "write a deployment yaml for the mongodb image"}'
+
+    grep -e "\"$h\":\"XXXXX\"" "$OLS_LOGS" || { echo "header $h was not redacted"; rc=1; }
+  done
+fi
```

Comment:
writing tests directly into our bash script isn't sustainable or maintainable.

we've got other python-based tests that inspect the log output, @onmete has written a few, you can copy his examples.  You can mark the test for cluster-based testing only, if that makes it easier.

over time i'd like to see test-e2e-standalone.sh go away entirely, but for now:
1) it's a useful way for developers to do some local testing of the current code
2) some tests would be more difficult to write for running against OLS running in a cluster

but in general, testing against ols running in a cluster should be our preference as it reflects the way OLS actually works, and long term the goal for our e2e CI should be:

1) use the latest operator image to deploy the latest ols images to a cluster (where "latest" means last published, or built from the PR being tested, as applicable)
2) run our e2e against that env